bayes_opt/acquisition.py

```diff
@@ -198,15 +217,28 @@ def _acq_min(self, acq: Callable, bounds: np.ndarray, n_random=10_000, n_l_bfgs_
         if n_random == 0 and n_l_bfgs_b == 0:
             error_msg = "Either n_random or n_l_bfgs_b needs to be greater than 0."
             raise ValueError(error_msg)
+        x_min_r: NDArray[Float] | None
+        x_min_l: NDArray[Float] | None
+        min_acq_r: float
+        min_acq_l: float
         x_min_r, min_acq_r = self._random_sample_minimize(acq, bounds, n_random=n_random)
         x_min_l, min_acq_l = self._l_bfgs_b_minimize(acq, bounds, n_x_seeds=n_l_bfgs_b)
+        # Either n_random or n_l_bfgs_b is not 0 => at least one of x_min_r and x_min_l is not None
         if min_acq_r < min_acq_l:
             return x_min_r
         return x_min_l
 
+    @overload
+    def _random_sample_minimize(
+        self, acq: Callable[[NDArray[Float]], NDArray[Float]], bounds: NDArray[Float], n_random: Literal[0]
+    ) -> tuple[None, float]: ...
+    @overload
     def _random_sample_minimize(
-        self, acq: Callable, bounds: np.ndarray, n_random: int
-    ) -> tuple[np.ndarray, float]:
+        self, acq: Callable[[NDArray[Float]], NDArray[Float]], bounds: NDArray[Float], n_random: int
+    ) -> tuple[NDArray[Float] | None, float]: ...
+    def _random_sample_minimize(
+        self, acq: Callable[[NDArray[Float]], NDArray[Float]], bounds: NDArray[Float], n_random: int
+    ) -> tuple[NDArray[Float] | None, float]:
```

Comment:
Why is `@overload` necessary here? I guess you're trying to differentiate the version that uses `n_random=0` from the others?

[Per the docs](https://docs.python.org/3/library/typing.html#typing.overload), should the non-decorated function (L.239) be without type hints completely?

Reply:
You are correct that I used it to handle `n_random`.
But when I checked where it's used, it doesn't seem to be necessary.
(I guess it was needed when using generics to control the return value of `_acq_min`).

It doesn't matter though, as I plan to remove `overload` now,
`overload` has an impact when using `_random_sample_minimize`,
Without type hints, the type of the parameter during implementation of `_random_sample_minimize` is analyzed as `unknown`.